### PR TITLE
UF-10996: Add service logic for seding webmessages

### PR DESCRIPTION
**Pull request by ronnberg**
## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/integration-test/java/se/sundsvall/supportmanagement/apptest/ErrandTimeMeasurementsIT.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import org.springframework.data.domain.Example;
 import org.springframework.test.context.jdbc.Sql;
 
+import org.springframework.transaction.annotation.Transactional;
```
Comment (foobar-1337) [inline src/integration-test/java/se/sundsvall/supportmanagement/apptest/ErrandTimeMeasurementsIT.java:24]: Unused

File: src/test/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapperTest.java [hunk 2]
```diff
@@ -3,26 +3,37 @@
...
+import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
 
+import org.zalando.problem.Problem;
 import se.sundsvall.supportmanagement.api.model.communication.EmailAttachment;
 import se.sundsvall.supportmanagement.api.model.communication.EmailRequest;
 import se.sundsvall.supportmanagement.api.model.communication.SmsRequest;
+import se.sundsvall.supportmanagement.api.model.communication.WebMessageAttachment;
+import se.sundsvall.supportmanagement.api.model.communication.WebMessageRequest;
+import se.sundsvall.supportmanagement.api.model.errand.ExternalTag;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapperTest.java:33]: Unused

File: src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java [hunk 3]
```diff
@@ -88,6 +132,20 @@ static EmailAttachment toEmailAttachment(final AttachmentEntity attachment) {
...
+		try {
+			byte[] bytes = IOUtils.toByteArray(attachment.getAttachmentData().getFile().getBinaryStream());
+			String encoded = Base64.getEncoder().encodeToString(bytes);
+
+			return new WebMessageAttachment()
+				.base64Data(encoded)
+				.mimeType(detectMimeType(attachment.getFileName(), bytes))
+				.fileName(attachment.getFileName());
+		} catch (SQLException | IOException e) {
+			return null;
```
[the remark this answers is not in the document]
Comment (cheezi0747) [reply src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java:145]: hmm

File: src/main/java/se/sundsvall/supportmanagement/service/mapper/CommunicationMapper.java [hunk 4]
```diff
@@ -19,7 +19,9 @@
 import se.sundsvall.supportmanagement.api.model.communication.CommunicationAttachment;
 import se.sundsvall.supportmanagement.api.model.communication.EmailAttachment;
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/service/mapper/CommunicationMapper.java:20]: Unused

File: src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (foobar-1337) [reply src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java:145]: hmm

[the remark this answers is not in the document]
Comment (ronnberg) [reply src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java]: mhm

**Review by cheezi0747 — APPROVED** [review]

File: src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java [hunk 5]
```diff
@@ -84,7 +131,23 @@ static EmailAttachment toEmailAttachment(final AttachmentEntity attachment) {
...
 				.name(attachment.getFileName());
 		} catch (SQLException | IOException e) {
-			return null;
+			LOG.error("Attachment mapping error", e);
+			throw Problem.valueOf(INTERNAL_SERVER_ERROR, String.format("Failed to map attachment with id '%s'", attachment.getId()));
+		}
+	}
+
+	static WebMessageAttachment toWebMessageAttachment(final AttachmentEntity attachment) {
+		try {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/service/mapper/MessagingMapper.java:140]: try with resources?

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**